src/ol/source/GeoTIFF.js

```diff
@@ -308,6 +309,7 @@ function getMaxForDataType(array) {
  * sources, one with 3 bands and {@link import("./GeoTIFF.js").SourceInfo nodata} configured, and
  * another with 1 band, the resulting data tiles will have 5 bands: 3 from the first source, 1 alpha
  * band from the first source, and 1 band from the second source.
+ * @property {object} [sourceOptions] Additional options to be passed to the underlying geotiff.js source.
```

Comment:
For the user it is not obvious what `object` to pass as `sourceOptions`. This should be replaced with a type defined by a `@typedef`, which lists and explains the available options. To keep the API lean, only the options that we think are necessary (e.g. setting request headers) should be listed there.

Reply:
@ahocevar Updated. Could you have a look?